pints/tests/test_nested_ellipsoid_sampler.py

```diff
@@ -0,0 +1,195 @@
+        self.assertRaisesRegex(
+            ValueError, 'exceed 1', sampler.set_ellipsoid_update_gap, 0.5)
+        self.assertRaisesRegex(
+            ValueError, 'exceed 1', sampler.set_ellipsoid_update_gap, 1)
+
+        # dynamic enlargement factor
+        self.assertTrue(not sampler.dynamic_enlargement_factor())
+        sampler.set_dynamic_enlargement_factor(1)
+        self.assertTrue(sampler.dynamic_enlargement_factor())
+
+        # alpha
+        self.assertRaises(ValueError, sampler.set_alpha, -0.2)
+        self.assertRaises(ValueError, sampler.set_alpha, 1.2)
+        self.assertEqual(sampler.alpha(), 0.2)
+        sampler.set_alpha(0.4)
+        self.assertEqual(sampler.alpha(), 0.4)
+
+        # initial phase
+        self.assertTrue(sampler.needs_initial_phase())
+        self.assertTrue(sampler.in_initial_phase())
+        sampler.set_initial_phase(False)
+        self.assertTrue(not sampler.in_initial_phase())
+        self.assertEqual(sampler.name(), 'Nested ellipsoidal sampler')
+
+    def test_ask_tell(self):
+        # Tests ask and tell
+
+        # test that ellipses are estimated
+        sampler = pints.NestedEllipsoidSampler(self.log_prior)
+        A1 = np.copy(sampler._A)
+        c1 = sampler._centroid
+        sampler.set_n_rejection_samples(100)
+        sampler.set_ellipsoid_update_gap(10)
+        for i in range(5000):
+            pt = sampler.ask(1)
+            fx = self.log_likelihood(pt)
+            sampler.tell(fx)
+        A2 = sampler._A
+        c2 = sampler._centroid
+        self.assertTrue(not np.array_equal(A1, A2))
+        self.assertTrue(not np.array_equal(c1, c2))
+
+        # test multiple points being asked and tell'd
+        sampler = pints.NestedEllipsoidSampler(self.log_prior)
+        pts = sampler.ask(50)
+        self.assertEqual(len(pts), 50)
+        fx = [self.log_likelihood(pt) for pt in pts]
+        proposed = sampler.tell(fx)
+        self.assertTrue(len(proposed) > 1)
+
+        # test multiple ask points after rejection samples
+        sampler = pints.NestedEllipsoidSampler(self.log_prior)
+        sampler.set_n_rejection_samples(10)
+        for i in range(100):
+            self.assertEqual(len(sampler.ask(20)), 20)
+
+    def test_dynamic_enlargement_factor(self):
+        # tests dynamic enlargement factor runs
+        sampler = pints.NestedController(self.log_likelihood,
+                                         self.log_prior)
+        sampler._sampler.set_dynamic_enlargement_factor(1)
+        sampler.set_log_to_screen(False)
+        ef1 = sampler._sampler.enlargement_factor()
+        sampler.run()
+        ef2 = sampler._sampler.enlargement_factor()
+        self.assertTrue(ef2 < ef1)
+
+    def test_sensitivities(self):
+        # tests whether sensitivities bit runs
+        sampler = pints.NestedController(self.log_likelihood,
+                                         self.log_prior)
+        # hacky but currently no samplers need sensitivities
```

Comment:
Are we expecting any to need them? Otherwise maybe comment out this functionality for now?

Reply:
Yes, they will, so can we keep it please?